### PR TITLE
GDS: fix Application unregistration in SQL DB for Winforms Server & fix private key storage in Pull Model GDS Client

### PR DESCRIPTION
**Pull request by romanett**
## Proposed changes

- GDS: fix Application unregistration in SQL DB for Winforms Server
- fix private key creation for New Certificate in Pull Model GDS Client

## Related Issues

- Fixes #

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds functionality)
- [ ] Test enhancement (non-breaking change to increase test coverage)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected, requires version increase of Nuget packages)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/OPCFoundation/UA-.NETStandard/blob/master/CONTRIBUTING.md) doc.
- [x] I have signed the [CLA](https://opcfoundation.org/license/cla/ContributorLicenseAgreementv1.0.pdf).
- [ ] I ran tests locally with my changes, all passed.
- [ ] I fixed all failing tests in the CI pipelines. 
- [ ] I fixed all introduced issues with CodeQL and LGTM.
- [ ] I have added tests that prove my fix is effective or that my feature works and increased code coverage.
- [ ] I have added necessary documentation (if appropriate).
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: Samples/GDS/Client/Controls/ApplicationCertificateControl.cs [hunk 1]
```diff
@@ -316,41 +313,48 @@ private async void CertificateRequestTimer_Tick(object sender, EventArgs e)
...
+                        CertificateIdentifier cid = new CertificateIdentifier() {
                             StorePath = m_application.CertificateStorePath,
                             StoreType = CertificateStoreIdentifier.DetermineStoreType(m_application.CertificateStorePath),
                             SubjectName = m_application.CertificateSubjectName.Replace("localhost", Utils.GetHostName())
                         };
 
                         // update store
-                        using (var store = CertificateStoreIdentifier.OpenStore(m_application.CertificateStorePath))
+                        ICertificateStore store;
+
```
Comment (mregen) [inline Samples/GDS/Client/Controls/ApplicationCertificateControl.cs:324]: TODO: support NoPrivateKey flag in CertStoreIdentifier

File: Samples/GDS/Client/Controls/ApplicationCertificateControl.cs [hunk 2]
```diff
@@ -316,41 +313,48 @@ private async void CertificateRequestTimer_Tick(object sender, EventArgs e)
 
                     if (!String.IsNullOrEmpty(m_application.CertificateStorePath) && !String.IsNullOrEmpty(m_application.CertificateSubjectName))
                     {
-                        CertificateIdentifier cid = new CertificateIdentifier()
-                        {
+                        CertificateIdentifier cid = new CertificateIdentifier() {
                             StorePath = m_application.CertificateStorePath,
                             StoreType = CertificateStoreIdentifier.DetermineStoreType(m_application.CertificateStorePath),
                             SubjectName = m_application.CertificateSubjectName.Replace("localhost", Utils.GetHostName())
```
Comment (mregen) [inline Samples/GDS/Client/Controls/ApplicationCertificateControl.cs:319]: there is a helper in Utils to replace the hostname

Comment (romanett) [reply Samples/GDS/Client/Controls/ApplicationCertificateControl.cs:319]: is Utils.ReplaceDCLocalhost okay or better to use Utils.ReplaceLocalhost to catch uncommon SubjectNames?

Comment (mregen) [reply Samples/GDS/Client/Controls/ApplicationCertificateControl.cs:319]: Utils.ReplaceDCLocalhost should be it

File: Samples/GDS/Client/Controls/ApplicationCertificateControl.cs (hunk 1 above)
Comment (romanett) [reply Samples/GDS/Client/Controls/ApplicationCertificateControl.cs]: done with preview, waiting for final release

**Review by mregen — APPROVED** [review]

**Comment by mregen** [comment]
Nuget versions mismatch, build error

**Comment by romanett** [comment]
> Nuget versions mismatch, build error

locally my build works fine, can you provide an error message?

**Comment by mregen** [comment]
see https://github.com/OPCFoundation/UA-.NETStandard-Samples/pull/603/checks?check_run_id=24548958539

**Pull request merged**